Modules/Sources/WordPressUI/Views/AdaptiveTabBarController.swift

```diff
@@ -0,0 +1,171 @@
+    private var filterBarContainer = UIView()
+
+    private let segmentedControl = UISegmentedControl()
+
+    private weak var viewController: UIViewController?
+
+    private var onSelectionChanged: ((Item) -> Void)?
+
+    private var isFilterBarHidden = false
+    private var lastContentOffset: CGFloat = 0
+    private var scrollViewObserver: NSKeyValueObservation?
+    private var traitObserver: NSObjectProtocol?
+
+    public var navigationItem: UINavigationItem?
+
+    public init() {
+        setupFilterBar()
+        setupSegmentedControl()
+    }
+
+    private func setupFilterBar() {
+        filterBarContainer.backgroundColor = .secondarySystemGroupedBackground
+        filterBarContainer.addSubview(filterBar)
+        filterBar.pinEdges(.top, to: filterBarContainer.safeAreaLayoutGuide, insets: UIEdgeInsets(.top, -filterBar.tabBarHeight))
+        filterBar.pinEdges([.horizontal, .bottom])
+
+        filterBar.addTarget(self, action: #selector(selectedFilterDidChange), for: .valueChanged)
+    }
+
+    private func setupSegmentedControl() {
+        segmentedControl.addTarget(self, action: #selector(segmentedControlValueChanged), for: .valueChanged)
+    }
+
+    public func configure(_ items: [Item], in viewController: UIViewController, onSelectionChanged: @escaping (Item) -> Void) {
+        self.items = items
+        self.selection = items.first
+        self.viewController = viewController
+        self.onSelectionChanged = onSelectionChanged
+
+        filterBar.items = items
+
+        segmentedControl.removeAllSegments()
+        for (index, item) in items.enumerated() {
+            segmentedControl.insertSegment(withTitle: item.localizedTitle, at: index, animated: false)
+        }
+        segmentedControl.selectedSegmentIndex = 0
+
+        setupTraitObserver(for: viewController)
+        refresh()
+    }
+
+    private func refresh() {
+        guard let viewController else { return }
+
+        if viewController.traitCollection.horizontalSizeClass == .regular {
+            filterBarContainer.removeFromSuperview()
+            (navigationItem ?? viewController.navigationItem).leftBarButtonItem = UIBarButtonItem(customView: segmentedControl)
+            viewController.additionalSafeAreaInsets = .zero
+        } else {
+            viewController.navigationItem.titleView = nil
+            viewController.view.addSubview(filterBarContainer)
+            filterBarContainer.pinEdges([.top, .horizontal])
+            viewController.additionalSafeAreaInsets = UIEdgeInsets(.top, filterBar.tabBarHeight)
+        }
+    }
+
+    private func setupTraitObserver(for viewController: UIViewController) {
+        if #available(iOS 17, *) {
+            traitObserver = viewController.registerForTraitChanges([UITraitHorizontalSizeClass.self]) { [weak self] (viewController: UIViewController, previousTraitCollection: UITraitCollection) in
+                self?.refresh()
+            }
+        }
```

Comment:
I guess it is okay to not handle pre-iOS 17, because we'll drop them soon.

Reply:
Yeah that was my thinking too. We'll likely drop it after iOS 26 release. I would advocate it for sure.